### PR TITLE
C++11: Avoid implicit copy/move assignment/construction

### PR DESCRIPTION
**Pull request by tbeu**
This is for #214. Hope I caught them all.

### CODE REVIEW

**Review by piponazo — DISMISSED** [review]
It looks great! Thanks!

**Review by piponazo — APPROVED** [review]

File: include/exiv2/datasets.hpp [hunk 1]
```diff
@@ -172,15 +172,12 @@ namespace Exiv2 {
         static const uint16_t Preview                = 202;
         //@}
 
-    private:
-        //! Prevent construction: not implemented.
-        IptcDataSets() {}
```
Comment (D4N) [inline include/exiv2/datasets.hpp]: This one was implemented. Bug or not?

File: src/tiffcomposite_int.hpp [hunk 2]
```diff
@@ -1030,13 +1024,10 @@ namespace Exiv2 {
...
         //@}
 
-    private:
-        //! @name NOT implemented
-        //@{
-        //! Assignment operator.
-        TiffSubIfd& operator=(const TiffSubIfd& rhs);
-        //@}
+    public:
+        TiffSubIfd& operator=(const TiffSubIfd& rhs) = delete;
```
Comment (D4N) [inline src/tiffcomposite_int.hpp]: Are the move assignment operators missing here intentionally?

File: include/exiv2/datasets.hpp (hunk 1 above)
Comment (piponazo) [reply include/exiv2/datasets.hpp]: I noticed the same, but if tests are passing ... I would not worry to much about it.

Comment (tbeu) [reply include/exiv2/datasets.hpp]: Yes, I also wondered. Probably a bug.

File: src/tiffcomposite_int.hpp (hunk 2 above)
Comment (tbeu) [reply src/tiffcomposite_int.hpp]: Addressed by 69e6b7e117243ac95288c76f6c608bfb0f640554.

**Comment by D4N** [comment]
@tbeu Can you resolve the conflicts & update this branch? Beside that: let's merge this!

**Comment by tbeu** [comment]
Ready to merge?

**Pull request merged**